model_analyzer/config/generate/model_variant_name_manager.py

```diff
@@ -21,95 +22,78 @@ class ModelVariantNameManager:
 
     def __init__(self):
 
-        # Stored as _variant_names[base_model_name][param_combo] = variant_name_string
-        self._variant_names = {}
+        # Dict of {model_config_name: model_config_dict}
+        self._model_config_dicts: Dict[str, Dict] = {}
 
-        # Stored as _model_name_index[base_model_name] = current_count_integer
-        self._model_name_index = {}
+        # Dict of {base_model_name: current_count_integer}
+        self._model_name_index: Dict[str, int] = {}
 
-    def get_model_variant_name(self, base_model_name, param_combo):
+    @classmethod
+    def from_dict(
+            cls,
+            model_variant_name_manager_dict: Dict) -> "ModelVariantNameManager":
+        model_variant_name_manager = ModelVariantNameManager()
+
+        model_variant_name_manager._model_config_dicts = model_variant_name_manager_dict[
+            '_model_config_dicts']
+        model_variant_name_manager._model_name_index = model_variant_name_manager_dict[
+            '_model_name_index']
+
+        return model_variant_name_manager
+
+    def get_model_variant_name(self, model_name: str, model_config_dict: Dict,
+                               param_combo: Dict) -> Tuple[bool, str]:
         """
         Given a base model name and a dict of parameters to be applied
-        to the base model config, return the name of the model variant
+        to the base model config, return the name of the model variant 
```

Comment:
the order of the english should match the order of the return values. Bool first.

Reply:
Done